### PR TITLE
Introduce python coverage CI

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

NA

## What does this implement/fix? Explain your changes.

Introduce python coverage CI based on optuna settings https://github.com/optuna/optuna/blob/master/.github/workflows/coverage.yml

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -33,6 +33,18 @@ dependencies = [
...
+[project.optional-dependencies]
+test = [
+    "coverage",
+    "pytest",
+]
+
+optional = [
+    "streamlit",
+    "boto3",
+    "moto[s3]",
```
Comment (c-bata) [inline pyproject.toml:45]: `moto[s3]` is a mock library for boto3, and is used by unit tests only. How about moving to the `test` category?

Comment (keisuke-umezawa) [reply pyproject.toml]: Fixed.

**Comment by keisuke-umezawa** [comment]
@c-bata 
Thank you for your review! I followed your review.

**Review by c-bata — APPROVED** [review]
LGTM! Thank you for your pull request.

**Pull request merged**